tools/antithesis-go-generator/catalog_output.go

```diff
@@ -0,0 +1,162 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
+	"text/template"
+)
+
+type GenInfo struct {
+	ExpectedVals      []*AntExpect
+	ExpectPackageName string
+}
+
+const GENERATED_SUFFIX = "_antithesis.go"
+
+func expect_output_file(module_name string) (*os.File, error) {
+	from_file := os.Getenv("GOFILE")
+	if from_file == "" {
+		from_file = module_name
+	}
+	dir_name, file_name := path.Split(from_file)
+	ext := path.Ext(file_name)
+	if len(ext) > 0 {
+		file_name = strings.TrimSuffix(file_name, ext)
+	}
+	generated_name := fmt.Sprintf("%s%s", file_name, GENERATED_SUFFIX)
+	output_file_name := path.Join(dir_name, generated_name)
+	fmt.Printf("Generated file: %q\n", output_file_name)
```

Comment:
If I have a file called foo.txt, will that become foo_antithesis_catalog.go? Are we sure we're only passing legit go files here, or are we doing the right thing if we're not?

In other words, the ``len(ext) > 0`` seems weird to me that we'd just continue if the extension isn't "go".

Reply:
Redone - as shown here:

```
// --------------------------------------------------------------------------------
// dest_path is always structured as a file path 
// that ends with the module name being instrumented
// the module name has been mangled by replacing path
// separators "/" and "\" with "_V_"
// 
// Example:
// If the 'dest_path' argument is:
//   "/home/johndoe/output/customer/nice.example.com_V_my_V_thing"
//
// Then 'dest_path' was composed of 
//   customerOutputDir: "/home/johndoe/output/customer"
//   moduleName: "nice.example.com/my/thing"
//
// This gets split to:
//   dir_name: "/home/johndoe/output/customer"
//   file_name: "nice.example.com_V_my_V_thing"
//   ext: ""
//
// And finally:
//   generated_name: "nice.example.com_V_my_V_thing_antithesis_catalog.go" 
//   output_file_name: "/home/johndoe/output/customer/nice.example.com_V_my_V_thing_antithesis_catalog.go"
//
// --------------------------------------------------------------------------------
func expectOutputFile(dest_path string) (*os.File, error) {
	dir_name, file_name := path.Split(dest_path)
	generated_name := fmt.Sprintf("%s%s", file_name, GENERATED_SUFFIX)
	output_file_name := path.Join(dir_name, generated_name)
```